### ISSUE TITLE
az artifacts download universal fails

### ISSUE THREAD

**Issue by MoBattah** [Azure/azure-cli-extensions#724]
PS C:\Users\mo.battah\testdownload&gt; az artifacts universal download --organization "https://dev.azure.com/vistex/" --fee
d "Vistex.DevOps" --name "testahdpackage" --version "1.2.0" --path .
Universal Packages is currently in preview.
The command failed with an unexpected error. Here is the traceback:

sequence item 16: expected str instance, NoneType found
Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-dsvrkf2x\knack\knack\cli.py", line 206, in invoke
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-dsvrkf2x\azure-cli-core\azure\cli\core\commands\__init__.py", l
ine 560, in execute
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-dsvrkf2x\azure-cli-core\azure\cli\core\commands\__init__.py", l
ine 618, in _run_jobs_serially
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-dsvrkf2x\azure-cli-core\azure\cli\core\commands\__init__.py", l
ine 609, in _run_job
  File "C:\Users\mo.battah\.azure\cliextensions\azure-devops\azext_devops\dev\common\exception_handler.py", line 26, in
azure_devops_exception_handler
    reraise(*sys.exc_info())
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-dsvrkf2x\six\six.py", line 693, in reraise
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-dsvrkf2x\azure-cli-core\azure\cli\core\commands\__init__.py", l
ine 588, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-dsvrkf2x\azure-cli-core\azure\cli\core\commands\__init__.py", l
ine 297, in __call__
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-dsvrkf2x\azure-cli-core\azure\cli\core\__init__.py", line 461,
in default_command_handler
  File "C:\Users\mo.battah\.azure\cliextensions\azure-devops\azext_devops\dev\artifacts\universal.py", line 55, in downl
oad_package
    return artifact_tool.download_universal(organization, feed, name, version, path, file_filter)
  File "C:\Users\mo.battah\.azure\cliextensions\azure-devops\azext_devops\dev\common\artifacttool.py", line 38, in downl
oad_universal
    return self.run_artifacttool(organization, args, "Downloading")
  File "C:\Users\mo.battah\.azure\cliextensions\azure-devops\azext_devops\dev\common\artifacttool.py", line 60, in run_a
rtifacttool
    proc = self._tool_invoker.run(command_args, new_env, initial_progress_message, _process_stderr)
  File "C:\Users\mo.battah\.azure\cliextensions\azure-devops\azext_devops\dev\common\external_tool.py", line 71, in run
    self.start(command_args, env)
  File "C:\Users\mo.battah\.azure\cliextensions\azure-devops\azext_devops\dev\common\external_tool.py", line 29, in star
t
    logger.debug("Running external command: %s", ' '.join(command_args))
TypeError: sequence item 16: expected str instance, NoneType found

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fa27bc14-1989-60eb-06ea-1270d9fbf0f1
* Version Independent ID: 1f391f98-ac28-ba9f-366a-447c1f9e4feb
* Content: [az artifacts universal](https://docs.microsoft.com/en-us/cli/azure/ext/azure-devops/artifacts/universal?view=azure-cli-latest#feedback)
* Content Source: [latest/docs-ref-autogen/ext/azure-devops/artifacts/universal.yml](https://github.com/MicrosoftDocs/azure-docs-cli/blob/live/latest/docs-ref-autogen/ext/azure-devops/artifacts/universal.yml)
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by plang85** [comment]
### **This is autogenerated. Please review and update as needed.**

## Describe the bug

**Command Name**
`az artifacts universal download
Extension Name: azure-devops. Version: 0.9.0.`

**Errors:**
```
sequence item 16: expected str instance, NoneType found
Traceback (most recent call last):
lib\site-packages\knack\cli.py, ln 206, in invoke
    cmd_result = self.invocation.execute(args)
cli\core\commands\__init__.py, ln 523, in execute
    raise ex
cli\core\commands\__init__.py, ln 581, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli\core\commands\__init__.py, ln 572, in _run_job
    cmd_copy.exception_handler(ex)
...
azext_devops\dev\common\external_tool.py, ln 71, in run
    self.start(command_args, env)
azext_devops\dev\common\external_tool.py, ln 29, in start
    logger.debug("Running external command: %s", ' '.join(command_args))
TypeError: sequence item 16: expected str instance, NoneType found
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az artifacts universal download --organization {} --feed {} --name {} --version {} --path {}`

## Expected Behavior

## Environment Summary
```
Windows-7-6.1.7601-SP1
Python 3.7.3
Shell: cmd.exe

azure-cli 2.0.63

Extensions:
azure-devops 0.9.0

```
## Additional Context

**Comment by corppeon** [comment]
I see this too

**Comment by RelNinetyPercent** [comment]
I believe this is because the `file-filter` parameter's constructor is being used incorrectly when it's not provided. In other words, it's a required parameter but the documentation is incorrect. If you add `file-filter '*'` the download works for me.

Also, the `detect` values are actually {false, true} in the latest cli/extension and not {off, on} as documented above.

**Comment by RelNinetyPercent** [comment]
I would guess the issue is here where None can still cause the argument to be added.
https://github.com/Azure/azure-devops-cli-extension/blob/master/azure-devops/azext_devops/dev/common/artifacttool.py#L38

**Comment by atbagga** [comment]
Fixed in https://github.com/Azure/azure-devops-cli-extension/pull/643/files
Released in Azure-devops version 0.10.0.

The issue can be closed now.
